### PR TITLE
fri16feb - 1

### PR DESCRIPTION
**Pull request by atoulous**

### CODE REVIEW

File: src/server/classes/Player.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * Class Player
  *
- * @param constructor {Object} - id, socket
+ * @param constructor {Object} - socketId, name (optional)
```
Comment (atoulous) [inline src/server/classes/Player.js]: need update

File: src/server/classes/Piece.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
-import _ from 'lodash';
+import get from 'lodash/get';
```
Comment (thifranc) [inline src/server/classes/Piece.js:1]: thanks

Comment (atoulous) [reply src/server/classes/Piece.js:1]: tkt bb

File: src/server/webSocket/handlePlayerSocket.js [hunk 3]
```diff
@@ -14,10 +14,12 @@ export default async function (data) {
...
   switch (path) {
     case '/new': {
-      console.log('new piece nedded', data);
+      console.log('new piece needed', data);
 
       const { room } = data;
-      const newPlayer = new Player();
+      const socketId = io.id;
+
+      const newPlayer = new Player({ room, socketId });
```
Comment (thifranc) [inline src/server/webSocket/handlePlayerSocket.js]: why new player ?
data should have a roomId + websocketId to have tracking info maybe (like which player asked for piece (help scoring), and which room(=game) to send new piece)

**Review by thifranc — COMMENTED** [review]
new Piece logic has some flaws

**Pull request merged**